contracts/src/v0.8/keystone/CapabilityRegistry.sol

```diff
@@ -107,21 +107,28 @@ contract CapabilityRegistry is OwnerIsCreator, TypeAndVersionInterface {
     bytes config;
   }
 
+  struct DONNodeCapabilityConfig {
+    /// @notice The set of p2pIds of nodes that belong to this DON. A node (the same
+    // p2pId) can belong to multiple DONs.
+    EnumerableSet.Bytes32Set nodes;
+    /// @notice The set of capabilityIds
+    EnumerableSet.Bytes32Set capabilityIds;
+    /// @notice Mapping from hashed capability IDs to configs
+    mapping(bytes32 capabilityId => bytes config) capabilityConfigs;
+  }
+
   /// @notice DON (Decentralized Oracle Network) is a grouping of nodes that support
   // the same capabilities.
   struct DON {
     /// @notice Computed. Auto-increment.
     uint32 id;
+    /// @notice The number of times the DON's capability was configured
+    uint32 configCount;
     /// @notice True if the DON is public.  A public DON means that it accepts
     /// external capability requests
     bool isPublic;
-    /// @notice The set of p2pIds of nodes that belong to this DON. A node (the same
-    // p2pId) can belong to multiple DONs.
-    EnumerableSet.Bytes32Set nodes;
-    /// @notice The set of capabilityIds
-    EnumerableSet.Bytes32Set capabilityIds;
-    /// @notice Mapping from hashed capability IDs to configs
-    mapping(bytes32 capabilityId => bytes config) capabilityConfigs;
+    /// @notice Mapping of config counts to configurations
+    mapping(uint32 configCount => DONNodeCapabilityConfig donConfig) nodeCapabilityConfigs;
```

Comment:
Thinking that it may also make sense to have separate `configCount`s for the different configs so that we don't have to repopulate everything when updating.